src/execution/compiler/operator/hash_aggregation_translator.cpp

```diff
@@ -387,6 +387,7 @@ void HashAggregationTranslator::UpdateAggregates(WorkContext *context, FunctionB
 
   // Advance aggregate.
   AdvanceAggregate(function, agg_payload, agg_values);
+  FeatureArithmeticRecordAdd(function, context->GetPipeline(), GetTranslatorId(), codegen->Const32(1));
```

Comment:
I am trying to convince myself that this is correct.

Reply:
Emm I'm surprised that we even need to do this. I think in theory we just need to multiply the arithmetic features by `num_agg_inputs_` (which you already did). By default each OU has a [`num_rows` = 1](num_agg_inputs_), and the arithmetic OUs will be invoked by `num_agg_inputs_` times. Adding another counter does not seem to be the best direction for me. Unless the arithmetics have a `num_rows` = 0 to begin with? Then that might be a bug in the operating unit recorder. Also, the multiplication of `num_agg_outputs_` in the `AGG_ITERATE` phase seems redundant to me...

Reply:
Removed, it was a different bug.

Could you elaborate on the redundancy?

Reply:
I see! In that case, wouldn't the `num_rows` and `cardinality` features for `AGG_BUILD` and `AGG_ITERATE` also be recorded many times instead of just one time in the end? (Though the eventual value is correct.) Maybe worth double-checking other counters.

For the redundancy - Basically I'm wondering would the `AGG_BUILD` and `AGG_ITERATE` have the same translator id or not? If they have the same translator id, then all the arithmetics would be multiplied twice in both of the two pipelines, right?